### PR TITLE
Made AntiSymmetric Tensor Canonical

### LINKED ISSUE #13707: make AntiSymmetricTensor canonical
There are two `_sortkey` routines in that file. One uses the label and the other doesn't leading to the issue raised by Matthew Chan:
```python
from sympy import symbols, Dummy
from sympy.physics.secondquant import AntiSymmetricTensor

v = symbols("v")
virtual_indices = ('c', 'd')
occupied_indices = ('k', 'l')

c, d = symbols(('c','d'), above_fermi=True,
                               cls=Dummy)
k, l = symbols(('k','l'), below_fermi=True,
                               cls=Dummy)

print(AntiSymmetricTensor(v, (k, l), (d, c)))
# Gives AntiSymTensor(kl,dc) or -AntiSymTensor(lk,dc) randomly between runs.
```
The following fixes the issue:
```diff
diff --git a/sympy/physics/secondquant.py b/sympy/physics/secondquant.py
index 384a296..0245425 100644
--- a/sympy/physics/secondquant.py
+++ b/sympy/physics/secondquant.py
@@ -205,20 +205,21 @@ def _sortkey(cls, index):
         FIXME: This is a bottle-neck, can we do it faster?
         """
         h = hash(index)
+        label = str(index)
         if isinstance(index, Dummy):
             if index.assumptions0.get('above_fermi'):
-                return (20, h)
+                return (20, label, h)
             elif index.assumptions0.get('below_fermi'):
-                return (21, h)
+                return (21, label, h)
             else:
-                return (22, h)
+                return (22, label, h)

         if index.assumptions0.get('above_fermi'):
-            return (10, h)
+            return (10, label, h)
         elif index.assumptions0.get('below_fermi'):
-            return (11, h)
+            return (11, label, h)
         else:
-            return (12, h)
+            return (12, label, h)
```

### PR DESCRIPTION
**Pull request by akshat14714**
As guided in the issue created, I made the subsequent changes in the file to fix it.

Fixes #13707 

@smichr  Please review.

### CODE REVIEW

**Comment by smichr** [comment]
A tests should be added, too.

**Comment by smichr** [comment]
The example from the issue page is a sufficient test.

**Comment by akshat14714** [comment]
@smichr Can you help me how to add the test for the issue?

**Comment by smichr** [comment]
In the folder with secondquant is a folder named tests. There should be a file named `test_secondquant.py`. At the end of the file make a new function named something like `def test_canonical_ordering_AntiSymmetricTensor():` and add the code from the issue. That function will then be called when tests are run. You can test it your self in an interactive session by entering the command `test('secondquant')`.

Leave 2 spaces before your function. Be sure that the last line of the file is a blank line.

**Comment by akshat14714** [comment]
@smichr Please review the code. I am not getting why it's the Tarvis build has failed.

**Comment by gxyd** [comment]
Travis errors are unrelated to your change. See https://github.com/sympy/sympy/issues/13730

File: sympy/physics/tests/test_secondquant.py [hunk 1]
```diff
@@ -1219,3 +1219,17 @@ def test_internal_external_pqrs_AT():
...
+    v = symbols("v")
+    virtual_indices = ('c', 'd')
+    occupied_indices = ('k', 'l')
+
+    c, d = symbols(('c','d'), above_fermi=True,
+                                   cls=Dummy)
+    k, l = symbols(('k','l'), below_fermi=True,
+                                   cls=Dummy)
+
+    print(AntiSymmetricTensor(v, (k, l), (d, c)))
```
Comment (smichr) [inline sympy/physics/tests/test_secondquant.py]: doctests will check what is printed, but tests require an assertion to be made. In this case I believe the canonical result will pass `assert AntiSymmetricTensor(v, (k, l), (d, c)) == -AntiSymmetricTensor(v, (l, k), (d, c))`...but check to see that this matches the output that was given by your print statement. Summary: replace your print statement with the assert statement and remove the comment (which demonstrated that the issue was valid).

**Comment by ylemkimon** [comment]
Please rebase on sympy/master. Thank you!

**Comment by akshat14714** [comment]
@ylemkimon rebasing on sympy/master shows up to date.

**Comment by ylemkimon** [comment]
Your branch seems to be [18 commits](https://github.com/akshat14714/sympy/compare/master...sympy:master) behind.

**Comment by akshat14714** [comment]
Done :+1:

**Comment by smichr** [comment]
Looks good, it's in.

**Pull request merged**